### DISCUSSION TITLE
Bug in on("click") on IOS

### DISCUSSION THREAD

**Discussion by alikalik9** [zauberzeug/nicegui#1748] [General]
I have a strange issue where my on click event is not firering on IOS devices. 
This is the minimal example:

```py
ui.label("hi").classes('lg:hidden').on("click", lambda:ui.open("https://www.google.de/", new_tab=True))
```
I tried different versions of the code. Tried a button with the built in on_click parameter, I also tried replacing the ui.open function with a custom javascript one. No success.

I have no issues whatsoever on Android, Windows or MacOS.

**Comment by falkoschindler** [comment]
It sounds like there's a communication problem between frontend and backend.
Can you, please, verify with the following code:

```py
@ui.page('/')
def page():
    async def get_date():
        print(await ui.run_javascript('Date()'))
    ui.timer(1.0, get_date)
```

This should request the current time from the client, without the need for a user interaction. This should let use narrow down the problem.

Besides that: What iOS version are you working on?

**Comment by alikalik9** [comment]
I tried the code and it works fine. In the meantime I was able to identify the cause for the behaviour, it is the 'new_tab' parameter. If it's set to "True" IOS ignores the link completely. If I set it to "False" it works. Seems like IOS is prohibiting the opening of a new tab window.
I can't check the IOS Version of the Iphone I testet it on right now (though it should be the newest one). For the other device it is Ipad OS 16.6.1.

**Comment by falkoschindler** [comment]
Ok, I think I understand: Safari seems to block new tabs by default, at least if it isn't a result of a user action. And since we're sending the click event to the server and the open command back to the client, it technically isn't a user action anymore. I could reproduce this behavior with Safari on Mac OS as well as on iOS.

This code allows to test it more explicitely:
```py
ui.button('_blank', on_click=lambda: ui.run_javascript('window.open("https://www.google.de/", "_blank")'))
ui.button('_self', on_click=lambda: ui.run_javascript('window.open("https://www.google.de/", "_self")'))
```

To workaround this limitation you can either
- omit the `new_tab` parameter to open the page in the same tab, or
- use a `ui.link` instead of a button, which is allowed to open new tabs because it is directly linked to a user action.

**Comment by alikalik9** [comment]
Thank you for checking, ui.link works fine for my case. Maybe sometimes it will be a little bit more work than just ui.open, because afaik ui.link automatically formats everything it wraps as underlined blue, but I'm sure you can revert that with css.
Maybe a hint in the documentation for ui.open about the limitation would be good for future users.

**Reply by falkoschindler** [reply]
Done: https://github.com/zauberzeug/nicegui/commit/116bb345d0ef0d89bdffb3e339df0c8c41f8400b ✔️
